service/api_logic/teams_logic.py

```diff
@@ -8,27 +12,94 @@
 RESPONSE_KEY = "response"
 
 def get_teams(session):
-    blob_indexes = get_all_blob_indexes_from_db(session, TEAMS_JSON)
-    result = get_blob_data_for_all_sports(session, blob_indexes)
-    data = json.loads(result)
-    filtered_data = []
+    try:
+        blob_indexes = get_all_blob_indexes_from_db(session, TEAMS_JSON)
+        result = get_blob_data_for_all_sports(session, blob_indexes)
+        data = json.loads(result)
+        filtered_data = []
 
-    for teams_data in data:
-        processed_data = process_blob_data(teams_data)
-        if processed_data:
-            filtered_data.append(processed_data)
-    return filtered_data, 200
+        for teams_data in data:
+            processed_data = process_blob_data(teams_data)
+            if processed_data:
+                filtered_data.append(processed_data)
+        return filtered_data, 200
+    except OperationalError:
+        raise DatabaseConnectionError()
+    except Exception:
+        raise Exception
 
 
 def process_blob_data(sport_data):
-    blob_name = sport_data.get("blob_name")
     team = sport_data.get(DATA_KEY, {}).get(RESPONSE_KEY, [])
     sport = sport_data.get(SPORT_KEY)
     if team:
         return {
-            "sport": sport,
-            #"blob_name": blob_name,
+            SPORT_KEY: sport,
             TEAM_KEY: team,
         }
+    return None
 
-    return None
\ No newline at end of file
+
+def get_teams_sport(session, sport_type):
+    try:
+        sport = get_sport_by_name(session, sport_type)
+    except SportNotFoundError as e:
+        return get_error_response({"error": e.message}, 404)
+    try:
+        blob_indexes = get_all_blob_indexes_from_db(session, TEAMS_JSON)
+        result = get_blob_data_for_all_sports(session, blob_indexes)
+        data = json.loads(result)
+        filtered_data = []
+        for teams_data in data:
```

Comment:
while it's not a mistake to write code like this, it is still not "Python-like".

use your programming language techniques. For example the code above could be rewritten to:

filtered_data = [
    processed_data
    for teams_data in data
    if sport_type == teams_data.get(SPORT_KEY) and (processed_data := process_blob_data(teams_data))
]

should work, but, of course, I didn't test it :)